### PR TITLE
LE Audio: BAP API rename

### LINKED ISSUE #45622: LE Audio: Update BAP API with new naming scheme
**Is your enhancement proposal related to a problem? Please describe.**
Apply the new naming scheme from https://github.com/zephyrproject-rtos/zephyr/issues/40920 to the MICP implemenation and API.

For BAP it will mostly be to omit the `bt_audio` prefix as well as splitting audio.h into role-specific files, and probably also creating a stream.h 

All values related to generic audio assigned numbers, etc. can remain in audio.h 

**Describe the solution you'd like**
N/A

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
See individual commits

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/45622

There are still several outstanding things that should have some attention, most related to `bt_codec`, and whether it should be `bt_audio_codec` or stay as `bt_codec`, but be moved from `audio.h`. 

Note for reviewers: I suggest to keep your focus at the public API files. There are no functionality changes, and the changes in all files are merely renames.

### CODE REVIEW

**Review by larsgk — CHANGES_REQUESTED** [review]
Only a few comments - otherwise LGTM. Was the changes tested with shell in a live setup?

File: subsys/bluetooth/audio/bap_unicast_client.c [hunk 1]
```diff
@@ -619,26 +597,23 @@ static void unicast_client_ep_qos_state(struct bt_audio_ep *ep,
 
 	ep->cig_id = qos->cig_id;
 	ep->cis_id = qos->cis_id;
-	(void)memcpy(&stream->qos->interval, sys_le24_to_cpu(qos->interval),
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/audio/bap_unicast_client.c:622]: > Did the formatting rules change? There seem to be quite a few re-indentation/line break diffs in this PR (probably fine)

Yes, the TSC recently changes the column width from 80 (recommended) and 100 (maximum) to just being 100 (maximum). Any such line changes has been done by `clang-format` in this PR, and I'll be happy to switch them back if requested

File: subsys/bluetooth/audio/bap_unicast_server.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/bap.h>
+
+extern const struct bt_bap_unicast_server_cb *unicast_server_cb;
+
+int bt_unicast_server_reconfig(struct bt_bap_stream *stream, const struct bt_codec *codec);
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/audio/bap_unicast_server.h:14]: Yes! Thanks, must've missed that one

[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/audio/bap_unicast_server.h]: Fixed this and several related ones for both the unicast client and server.

**Review by larsgk — DISMISSED** [review]

**Review by MariuszSkamra — DISMISSED** [review]
LGTM

**Comment by carlescufi** [comment]
@Thalley please rebase

**Comment by Thalley** [comment]
Rebased to solve merge conflicts from https://github.com/zephyrproject-rtos/zephyr/pull/55368 and https://github.com/zephyrproject-rtos/zephyr/pull/55367

**Comment by Thalley** [comment]
Rebased to solve merge conflicts from https://github.com/zephyrproject-rtos/zephyr/pull/53564

File: subsys/bluetooth/audio/bap_stream.c [hunk 3]
```diff
@@ -63,25 +64,22 @@ void bt_audio_codec_to_iso_path(struct bt_iso_chan_path *path,
...
 				   const struct bt_codec_qos *codec_qos)
 {
 	io->sdu = codec_qos->sdu;
 	io->phy = codec_qos->phy;
 	io->rtn = codec_qos->rtn;
 }
-#endif /* CONFIG_BT_AUDIO_UNICAST_CLIENT ||
-	* CONFIG_BT_AUDIO_BROADCAST_SOURCE ||
-	* CONFIG_BT_AUDIO_BROADCAST_SINK
+#endif /* CONFIG_BT_BAP_UNICAST_CLIENT ||                                                          \
```
Comment (larsgk) [inline subsys/bluetooth/audio/bap_stream.c:76]: Something looks a bid odd on the auto-formatting here

**Review by larsgk — APPROVED** [review]
Besides a few places with slight odd formatting (probably the new formatter), I didn't catch any thorns since the last review.  LGTM

File: subsys/bluetooth/audio/bap_stream.c (hunk 3 above)
Comment (Thalley) [reply subsys/bluetooth/audio/bap_stream.c:76]: Hmm, besides `clang-format` adding the `\` it doesn't seem odd to me.

**Review by MariuszSkamra — APPROVED** [review]

**Pull request merged**